### ISSUE TITLE
Traefik does not reload changes in Consul Catalog tags

### ISSUE THREAD

**Issue by trondhindenes** [containous/traefik#2370]
*bug*

### What did you do?
Traefik 1.4.2 does not seem to gracefully reload changes to Consul Catalog tags (which again drives the frontend/backend configuration of Traefik. This can be observed by:

1. Configure a service with a setting such as circuit breaker:
```json
        "tags": [
            "traefik.frontend.rule=Host:stuff.stuff.com",
            "traefik.frontend.entryPoints=http",
            "traefik.enable=true",
            "traefik.backend.circuitbreaker=NetworkErrorRatio() > 0.5"
        ]
```

If all is set up correctly, Traefik should populate frontend/backend rules for the service.

In order to see the issue, change a setting in the tags, for example to:
```json
        "tags": [
            "traefik.frontend.rule=Host:stuff.stuff.com",
            "traefik.frontend.entryPoints=http",
            "traefik.enable=true",
            "traefik.backend.circuitbreaker=ResponseCodeRatio(500, 600, 0, 600) > 0.5"
        ]
```

On the node offering the service, issue a consul reload in order to pick up the changes.
If using the Consul Web UI, the ui should now show the updated tags.

However, Traefik still uses the config from the first tag array, and has not updated. When testing the same with Traefik 1.3.8 it correctly updates itself, but on 1.4.2 it does not.

### What did you expect to see?
Traefik should update its config based on the updated Consul tags

### What did you see instead?
Traefik is using outdated Consul tags. Only if Traefik is restarted, it will grab the new tag set and configure itself correctly.

### Output of `traefik version`: (_What version of Traefik are you using?_)
```
Version:      v1.4.2
Codename:     roquefort
Go version:   go1.9.2
Built:        2017-11-02_01:47:46PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
traefikLogsFile = "/var/log/traefik/traefik.log"
checkNewVersion = false
ProvidersThrottleDuration = "5s"
logLevel = "DEBUG"


[retry]

[forwardingTimeouts]
dialTimeout = "5s"
responseHeaderTimeout = "10s"

defaultEntryPoints = ["http"]
[entryPoints]
    [entryPoints.http]
    address = ":80"

[web]
address = ":8080"

[consulCatalog]
constraints = ["tag==environment:uat","tag==lb_id:external-api01"]
watch = true
endpoint = "127.0.0.1:8500"
exposedByDefault = false
```

**Comment by trondhindenes** [comment]
FYI I've seen the same on 1.3.8 aswell. Node is gone in Consul, but Traefik takes a good while (+15mins) to discover the change, at least according to the traefik 8080 dashboard.

**Comment by traefiker** [comment]
Closed by #2389.
